include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,466 @@
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Svenja Mehringer <svenja.mehringer@fu-berlin.de>
+// ==========================================================================
+
+#ifndef SEQAN_INCLUDE_ARG_PARSE_VERSION_CHECK_H_
+#define SEQAN_INCLUDE_ARG_PARSE_VERSION_CHECK_H_
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <sys/stat.h>
+#include <time.h>
+#include <stdio.h>
+#include <regex>
+#include <future>
+#include <sys/types.h> 
+#include <utime.h>
+// ==========================================================================
+// Tags, Classes, Enums
+// ==========================================================================
+
+struct VersionCheck
+{
+    // ----------------------------------------------------------------------------
+    // Member Variables
+    // ----------------------------------------------------------------------------
+    std::string _url = "http://www.seqan.de/version_check/SeqAn_";
+    std::string _name;
+    std::string _version = "0.0.0";
+    std::string _program;
+    std::string _command;
+    std::string _website = "https://github.com/seqan/seqan/tree/master/apps";
+    std::future<bool> & _fut;
+
+#if defined(PLATFORM_WINDOWS)
+    std::string _path = std::string(getenv("UserProfile")) + "/.config/seqan";
+#else
+    std::string _path = std::string(getenv("HOME")) + "/.config/seqan";
+#endif
+    
+    //get system information
+#ifdef __linux
+    std::string _os = "Linux";
+#elif __APPLE__
+	std::string _os = "MacOS";
+#elif defined(PLATFORM_WINDOWS)
+	std::string _os = "Windows";
+#elif __FreeBSD__
+    std::string _os = "FreeBSD";
+#elif __OpenBSD__
+    std::string _os = "OpenBSD";
+#else
+    std::string _os = "unknown";
+#endif
+
+    // ----------------------------------------------------------------------------
+    // Constructors
+    // ----------------------------------------------------------------------------
+    VersionCheck(std::future<bool> & fut,
+                 std::string const & name,
+                 std::string const & version,
+                 std::string const & website):
+    _fut(fut)
+    {
+        _name = name;
+        if (!version.empty() &&
+            std::regex_match(version, std::regex("^[[:digit:]]\\.[[:digit:]]\\.[[:digit:]].*")))
+            _version = version.substr(0,5); // in case the git revision number is given take only version number
+        if(!website.empty())
+            _website = website;
+        _getProgram();
+        _updateCommand();
+    }
+
+    VersionCheck(VersionCheck const & rhs):
```

Comment:
I think the class should not be copy constructible if it holds a reference to a future. 
Otherwise it is possible to write to the same future from multiple threads without knowing.